packages/store-ui/src/molecules/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import type { PropsWithChildren } from 'react'
+
+import Button from '../../atoms/Button'
+import Input from '../../atoms/Input'
+
+interface ActionButtonProps {
+  actionOption: 'dec' | 'inc'
+  disabled: boolean
+  ariaLabel: string
+  onClick: (currentValue: number) => void
+}
+
+const ActionButton = ({
+  children,
+  actionOption,
+  disabled,
+  ariaLabel,
+  onClick,
+}: PropsWithChildren<ActionButtonProps>) => (
+  <Button
+    type="button"
+    data-store-button={actionOption}
+    onClick={(event) => onClick(parseInt(event.currentTarget.value, 10))}
```

Comment:
Does the currentTarget always have the `value` property? I'm asking because the button doesn't have `value,` but since you're using currentTarget, it can have because children can be any HTML element with `value`.

If you want to get the value from the button, change currentTarget to target.
[MDN Doc](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget)
```suggestion
    onClick={(event) => onClick(parseInt(event.target.value, 10))}
```